keps/sig-storage/1472-storage-capacity-tracking/README.md

```diff
@@ -795,79 +840,149 @@ enhancement:
 * **Are there any tests for feature enablement/disablement?**
   The e2e framework does not currently support enabling and disabling feature
   gates. However, unit tests in each component dealing with managing data created
-  with and without the feature are necessary. At the very least, think about
-  conversion tests if API types are being modified.
+  with and without the feature are necessary and will be added before
+  before the transition to beta.
 
 ### Rollout, Upgrade and Rollback Planning
 
-Will be added before the transition to beta.
-
 * **How can a rollout fail? Can it impact already running workloads?**
 
+A rollout happens in at least two phases:
+1. Updating the cluster so that the `CSIStorageCapacity` API is enabled in the apiserver
+   and the kube-scheduler uses that information *for drivers which have opted into this*.
+2. CSI driver installations get updated such that they produce `CSIStorageCapacity` objects
+   and enable usage of those objects in their `CSIDriver` object.
+
+If for whatever reason the first phase fails, the feature can be left
+disabled and the second phase can be skipped. The cluster the operates
+as before.
+
+If the second phase fails because a driver malfunctions or overloads
+the apiserver, then it can be rolled back and scheduling again happens
+without using storage capacity information.
```

Comment:
Re these two paragraphs - can you please add what failure modes potentially this particular feature is introducing?

As an example - when enabling the feature in kube-apiserver, what can potentially go wrong as a result of this feature.

Reply:
> For this particular example, enabling it just in kube-apiserver should not fail in any way. It's just a new type that isn't getting used. Sorry for my lack of imagination

Imagine that e.g. conversion functions are broken for the type and they are crashing kube-apiserver. Then if there was a preexisting object, kube-apiserver won't even start...

I would probably mentioned two things in this case:
- unexpected kube-apiserver restarts
- increased number of 5xx errors from CSIStorageCapacity-related calls

Reply:
Added